Adding Preliminary Schematic

Tejasgarg opened the pull request:
Currently added the following:
- Main microprocessor and its power, ground, and oscillator connections 
- ESP32C3 with its SPI, UART and Chip Select connections
- USB-C connections
- Voltage Regulator, Doubler, and Inverter 
- Boot and reset switch button controls
- User and Power LED
- Battery Backup system
- Serial Wire Debugger Connections
- Mounting Holes

The voltage and Battery systems are largely taken from the previous PSLab board schematic.
I will continue working on this to add the rest of the components to the schematic 

- fixes: #7

CloudyPadmal commented:
1. ESP32-C3 needs an external flash. It is missing in the schematic.
2. What is the idea behind all those numbering (bottom left in each block)?
3. What is the purpose of J9 (02x10 Connector)? Only 2 pins are used to link CH1 and CH2, with rest unconnected.

Tejasgarg commented:
> 1. ESP32-C3 needs an external flash. It is missing in the schematic.
> 2. What is the idea behind all those numbering (bottom left in each block)?
> 3. What is the purpose of J9 (02x10 Connector)? Only 2 pins are used to link CH1 and CH2, with the rest unconnected.

1. Right! I missed that
I will include an external flash for the ESP32-C3
2. I used the numbering simply to number individual blocks in the schematic. Should it be removed?
3. I plan to add more connections for the Logic Analyser, Multimeter and GPIO to that header

CloudyPadmal commented:
Most of the components do not have any manufacturer's part numbers from LCSC. We agreed to add a **MPN** column for each part with such details. Please update all the parts with part numbers as you are adding them into the schematic, starting from all the existing parts. Use LCSC part numbers.

- [ ] There are 20 Ohm resistors connected in series with SPI line for ESP. What is the purpose of those resistors and 20 Ohm value?
- [ ] PCB_Antenna symbol is misleading. Use standard symbols.
- [ ] ESP Flash chip is just a 'Flash 3v3'. What is the part number?
- [ ] Net `VDD_SPI` is not connected to anywhere.
- [ ] Why a 499 Ohm resistor is used in UART1_TX connection to ESP?
- [ ] I would rename `SPI_TRIGGER` signal to something meaningful like `ESP_DATA_READY`.
- [ ] Why is there an inductor connected to ESP crystal?
- [ ] Mounting Holes block make the sheet wider. Move it down to a vacant space.
- [ ] There is an LED block with LEDs. Move the LEDs in Battery Backup System to that block.

AsCress commented:
> ![Schematic Preview](https://raw.githubusercontent.com/fossasia/pslab-mini-hardware/schematic-previews/schematic-8:e6494a5.png)

@CloudyPadmal Congratulations xD 😂 !

Tejasgarg commented:
> Most of the components do not have any manufacturer's part numbers from LCSC. We agreed to add a **MPN** column for each part with such details. Please update all the parts with part numbers as you are adding them into the schematic, starting from all the existing parts. Use LCSC part numbers.

I have updated the Datasheet, Manufacturer part number as well as Part Number for each component in the schematic.
Now each component reflects updated LCSC part number.

> 
> * [x]  There are 20 Ohm resistors connected in series with SPI line for ESP. What is the purpose of those resistors and 20 Ohm value?

Updated this to 0 ohm resistor as discussed

> * [x]  PCB_Antenna symbol is misleading. Use standard symbols.

Updated the symbol to a more generic symbol

> * [x]  ESP Flash chip is just a 'Flash 3v3'. What is the part number?

Renamed the flash symbol to the actual part number(W25Q32JV)

> * [x]  Net `VDD_SPI` is not connected to anywhere.
> * [x]  Why a 499 Ohm resistor is used in UART1_TX connection to ESP?

Removed the resistor as discussed.

> * [x]  I would rename `SPI_TRIGGER` signal to something meaningful like `ESP_DATA_READY`.

Updated.
> * [x]  Why is there an inductor connected to ESP crystal?

Updated this to a 0 ohm resistor as discussed.

> * [x]  Mounting Holes block make the sheet wider. Move it down to a vacant space.

Updated.
> * [x]  There is an LED block with LEDs. Move the LEDs in Battery Backup System to that block.

Updated.

I also went ahead and updated the BNC connector to a more sleek one that will be more appropriate for our board and consume less space overall.
SInce there was no footprint for it, I also went ahead and created a new footprint for it and added that to the PR as well.

Also added the Square Wave pin both the STM board as well as external connectors as discussed. :)

CloudyPadmal commented:
The schematic looks complete now.

@bessman , could you please have a look at the [schematic](https://raw.githubusercontent.com/fossasia/pslab-mini-hardware/schematic-previews/schematic-8:0fa54c8.png) to check if the connectors are in order? Thanks!

After that, let's merge this to move forward.

bessman commented:
Looks good overall. A few questions/comments:

1. For my understanding, what it the purpose of the 200 Ohm series resistors on the LA1, LA2, and FQY pins?
2. Rather than passthrough the MCU pin-names like PB8, PB10, etc., I would prefer GPIO0, GPIO1, GPIO2, etc. for generic pins.
3. Perhaps we should take care to expose the ADC-connected pins as GPIO. These are:

ADC Channel | Pin | Current function
-- | -- | --
INP0 | PA0 | CH1
INP1 | PA1 | CH2
INP2 | PF11 | NC
INP3 | PA6 | SPI1_MISO
INP4 | PC4 | FCY
INP5 | PB1 | CS1
INP6 | PF12 | NC
INP7 | PA7 | SPI1_MOSI
INP8 | PC5 | LA1
INP9 | PB0 | USER_LED
INP10 | PC0 | VOL
INP11 | PC1 | RES
INP12 | PC2 | CAP
INP13 | PC3 | CAP
INP14 | PA2 | UART2_TX
INP15 | PA3 | UART2_RX
INP18 | PA4 | NC
INP19 | PA5 | SPI1_SCK

PF11 and PF12 are ADC1 only.

Most of these pins are already spoken for or made available. The possible candidates I see are:

- PA4 (currently NC) could be a GPIO pin
- FQY, LA1, and LA2 could perhaps use other pins, since they only need digital input functionality.
- Same for USER_LED, which only needs digital output functionality.

bessman commented:
Oh, one more thing: The boot pin is not correct, I think. The button enables users to pull it high, but this will cause the MCU to enter the on-chip bootloader, NOT our OpenBLT-based bootloader. The on-chip bootloader should only be used during development, and we don't use it at all.

I think we don't need a hardware button to enter the bootloader. The way the bootloader works right now is that it waits for a stay-in-boot command for about a second after reset, which I think is sufficient. If necessary, we can also add a drop-to-bootloader command from the main application.

Tejasgarg commented:
> 1. For my understanding, what it the purpose of the 200 Ohm series resistors on the LA1, LA2, and FQY pins?

The 200 ohm resistors are kept for the purpose of input protection, keeping it similar to the previous design of the pslab board.

> 2. Rather than passthrough the MCU pin-names like PB8, PB10, etc., I would prefer GPIO0, GPIO1, GPIO2, etc. for generic pins.

I agree, this will make it simpler and more uniform for the final board
I will switch the naming to this convention.

> 3. Perhaps we should take care to expose the ADC-connected pins as GPIO (to let users use them as analog inputs). These are:
> 
> ADC Channel	Pin	Current function
> INP0	PA0	CH1
> INP1	PA1	CH2
> INP2	PF11	NC
> INP3	PA6	SPI1_MISO
> INP4	PC4	FCY
> INP5	PB1	CS1
> INP6	PF12	NC
> INP7	PA7	SPI1_MOSI
> INP8	PC5	LA1
> INP9	PB0	USER_LED
> INP10	PC0	VOL
> INP11	PC1	RES
> INP12	PC2	CAP
> INP13	PC3	CAP
> INP14	PA2	UART2_TX
> INP15	PA3	UART2_RX
> INP18	PA4	NC
> INP19	PA5	SPI1_SCK
> PF11 and PF12 are ADC1 only.
> 
> Most of these pins are already spoken for or made available. The possible candidates I see are:
> 
> * PA4 (currently NC) could be a GPIO pin
> * FQY, LA1, and LA2 could perhaps use other pins, since they only need digital input functionality.
> * Same for USER_LED, which only needs digital output functionality.

Absolutely!
I will switch up these pins to ensure exposing the ADC pins as GPIO 
We can also mark this in the silk screen on the board itself to ensure it is communicated that these pins are analog input GPIO

Tejasgarg commented:
> Oh, one more thing: The boot pin is not correct, I think. The button enables users to pull it high, but this will cause the MCU to enter the on-chip bootloader, NOT our OpenBLT-based bootloader. The on-chip bootloader should only be used during development, and we don't use it at all.
> 
> I think we don't need a hardware button to enter the bootloader. The way the bootloader works right now is that it waits for a stay-in-boot command for about a second after reset, which I think is sufficient. If necessary, we can also add a drop-to-bootloader command from the main application.

Right!
Thank you so much for pointing that out 
I had missed that 
You are right, on this board, we don't require a boot switch

We will be able to flash the bootloader directly from the serial debug and the firmware can be updated over USB 

I will go ahead and change this in the schematic, remove the boot switch and directly connect the boot0 to ground.

Tejasgarg commented:
I also went ahead and checked what Mr. Roland mentioned about the USB pins that were missing in the schematic. According to the datasheet for the USB-C receptacle:

<img width="911" height="319" alt="Screenshot 2025-08-06 140810" src="https://github.com/user-attachments/assets/e53b243a-e0e0-457d-b610-b986879b901d" /></br>

All four pins that were missing in the schematic are already physically connected, so I don’t think it should cause any issues

Tejasgarg commented:
> That's interesting if we can make do without a Boot switch. I will look a bit more into this [AN2606](https://www.st.com/resource/en/application_note/an2606-introduction-to-system-memory-boot-mode-on-stm32-mcus-stmicroelectronics.pdf). I'm a bit curious as to how we can burn the OpenBLT on a fresh MCU.
> 
> Regarding the series resistors on LA paths, these are not for protection, but for something called 'series termination'. The required resistance (200 for PSLab V6) needs to be adjusted after simulating the final layout.

Oh, I wasn't aware of that and assumed that since they are at the input, they must be for the input current protection.
Thank you so much for giving me insight on that.
I will read more about this and try implementing it as you mentioned.

CloudyPadmal commented:
There's nothing to explicitly implement. Besides KiCAD doesn't have inbuilt SI simulation capabilities yet. So I will take care of it.

Tejasgarg commented:
Oh, ok, thanks!
